### DISCUSSION TITLE
Accessing currentSelection$, currentBlockType$, etc..

### DISCUSSION THREAD

**Discussion by bloodlinealpha** [mdx-editor/editor#363] [Q&A]
Hello!

I was wondering if someone could help or explain to me how to use the [Core API](https://mdxeditor.dev/editor/api). Specifically I am trying to read the [currently selected text](https://mdxeditor.dev/editor/api/variables/currentSelection_) and replace it with new text. I have went through every example and it is not entirely clear how to access these items.

Here is my current implementation:

```
<MDXEditor 
    ref={mdxRef}
    markdown={'# Hello, world!'}
    onChange={(md) => console.log('change', { md })}
    plugins={[
        toolbarPlugin({
            toolbarContents: () => (
            <>
                <UndoRedo />
                <Separator />
                <BoldItalicUnderlineToggles />
                <ListsToggle />
                <CodeToggle />
                <Separator />
                <BlockTypeSelect />
                <CreateLink />
                <Separator />
            </>
            )
        }),
        listsPlugin(),
        headingsPlugin(),
        linkPlugin(),
        linkDialogPlugin(),
        thematicBreakPlugin(),
        frontmatterPlugin(),
        markdownShortcutPlugin(),
        ]}
  />

```
Thanks!

**Comment by petyosi** [comment]
You can access all internal state parts through the plugin interface. Check [this help article](https://mdxeditor.dev/editor/docs/extending-the-editor) for more details.
